File: server/src/main/java/com/fluffy/exam/domain/dto/ExamDetailSummaryDto.java
package com.fluffy.exam.domain.dto;

import com.fluffy.exam.domain.ExamStatus;
import com.querydsl.core.annotations.QueryProjection;
import java.time.LocalDateTime;

public record ExamDetailSummaryDto(
        Long id,
        String title,
        String description,
        ExamStatus status,
        AuthorDto author,
        Long questionCount,
        Long likeCount,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {

    @QueryProjection
    public ExamDetailSummaryDto {
    }
}

File: server/src/main/kotlin/com/fluffy/exam/domain/dto/ExamDetailSummaryDto.kt
package com.fluffy.exam.domain.dto

import com.fluffy.exam.domain.ExamStatus
import java.time.LocalDateTime

data class ExamDetailSummaryDto(
    val id: Long,
    val title: String,
    val description: String,
    val status: ExamStatus,
    val author: AuthorDto,
    val questionCount: Long,
    val likeCount: Long,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
)